src/agent/src/watcher.rs

```diff
@@ -290,11 +297,9 @@ impl SandboxStorages {
                         logger,
                     )
                     .mount()?;
-                }
-                Ok(count) => {
-                    debug!(logger, "watched {} files", count);
-                }
-            };
+                    Ok::<usize, anyhow::Error>(0)
```

Comment:
`Ok(())`

Reply:
The [`or_else` for futures](https://docs.rs/futures/0.3.15/futures/future/trait.TryFutureExt.html#method.or_else) runs a closure in the error case which Ok type must match the future it applies to (`scan` in our case). So we must return an `Ok(usize)` simply doing `Ok(0)` is not enough and the compiler fails to infer the Error type of our closure (It must infer because we `await?;` on the `or_else`. See https://rust-lang.github.io/async-book/07_workarounds/02_err_in_async_blocks.html. So we need to be explicit on both type and use the turbofish notation...

Reply:
oh, right. closures.. :)
I ended up with something like this instead:
```rust
        for entry in self.0.iter_mut().filter(|e| e.watch) {
            if let Err(e) = entry.scan(logger).await {
                // ...
            }
        }
```
But that doesn't make too much difference, except that `Ok` return.